File: app/src/main/java/me/imzack/app/ender/receiver/BootCompleteReceiver.java
package me.imzack.app.ender.receiver;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import me.imzack.app.ender.util.SystemUtil;
import me.imzack.app.ender.model.database.DatabaseHelper;

import java.util.Map;

public class BootCompleteReceiver extends BroadcastReceiver {

    @Override
    public void onReceive(Context context, Intent intent) {
        Map<String, Long> reminderTimeMap = DatabaseHelper.getInstance().queryReminderTimeWithEnabledReminder();
        for (Map.Entry<String, Long> entry : reminderTimeMap.entrySet()) {
            SystemUtil.setReminder(entry.getKey(), entry.getValue());
        }
    }
}

File: app/src/main/java/net/zackzhang/app/end/receiver/BootCompleteReceiver.kt
package net.zackzhang.app.end.receiver

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent

import net.zackzhang.app.end.model.DataManager
import net.zackzhang.app.end.util.SystemUtil

class BootCompleteReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        val reminderTimeMap = DataManager.databaseHelper.queryReminderTimeWithEnabledReminder()
        for ((key, value) in reminderTimeMap) {
            SystemUtil.setReminder(key, value)
        }
    }
}
